com.unity.render-pipelines.universal/ShaderLibrary/Shadows.hlsl

```diff
@@ -350,17 +350,17 @@ half GetMainLightShadowFade(float3 positionWS)
     float3 camToPixel = positionWS - _WorldSpaceCameraPos;
     float distanceCamToPixel2 = dot(camToPixel, camToPixel);
 
-    half fade = saturate(distanceCamToPixel2 * _MainLightShadowParams.z + _MainLightShadowParams.w);
-    return fade;
+    float fade = saturate(distanceCamToPixel2 * float(_MainLightShadowParams.z) + float(_MainLightShadowParams.w));
+    return half(fade);
 }
 
 half GetAdditionalLightShadowFade(float3 positionWS)
 {
     float3 camToPixel = positionWS - _WorldSpaceCameraPos;
     float distanceCamToPixel2 = dot(camToPixel, camToPixel);
 
-    half fade = saturate(distanceCamToPixel2 * _AdditionalShadowFadeParams.x + _AdditionalShadowFadeParams.y);
-    return fade;
+    float fade = saturate(distanceCamToPixel2 * float(_AdditionalShadowFadeParams.x) + float(_AdditionalShadowFadeParams.y));
```

Comment:
Same for _AdditionalShaderFadeParams - is it half?

Reply:
Yes, `half4 _AdditionalShadowFadeParams;` in Shadows.hlsl